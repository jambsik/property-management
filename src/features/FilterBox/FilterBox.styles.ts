import { white } from './../../theme/colors';

export const filterBoxRootStyles = {
    width: '1000px',
    height: '90px',
    borderRadius: '36px',
    backgroundColor: white,
    display: 'flex',
};

export const filterBoxChildStyles = {
    width: '100%',
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginLeft: '36px',
    marginRight: '36px',
};
