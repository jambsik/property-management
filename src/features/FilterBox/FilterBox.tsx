import { Box } from '@mui/system';
import React, { ReactNode } from 'react';
import SearchIcon from '@mui/icons-material/Search';

import { filterBoxChildStyles, filterBoxRootStyles } from './FilterBox.styles';
import { IconButton } from '@mui/material';
import { Size } from '../../constants/Size';

interface FilterBoxProps {
    children: ReactNode;
}

const FilterBox = ({ children }: FilterBoxProps) => {
    return (
        <Box sx={filterBoxRootStyles}>
            <Box sx={filterBoxChildStyles}>
                {children}
                <IconButton aria-label="searcheable real estate" color="secondary">
                    <SearchIcon fontSize={Size.Large} />
                </IconButton>
            </Box>
        </Box>
    );
};

export default FilterBox;
