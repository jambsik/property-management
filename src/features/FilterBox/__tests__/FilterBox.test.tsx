import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import FilterBox from '../FilterBox';
describe('Filter Box component test', () => {
    const onSearchMock = jest.fn();
    const childComponent = <div>My child</div>;

    const pageObject = {
        iconButton: 'filter-box-button',
    };

    it('Should render component without errors', () => {
        const { container } = render(<FilterBox onSearch={onSearchMock}>{childComponent}</FilterBox>);
        expect(container).toMatchSnapshot();
    });

    it('Should dispatch action after click on icon button', () => {
        const { getByTestId } = render(<FilterBox onSearch={onSearchMock}>{childComponent}</FilterBox>);

        fireEvent.click(getByTestId(pageObject.iconButton));

        expect(onSearchMock).toHaveBeenCalledTimes(1);
    });

    it('Should validate child renderization', () => {
        const { container } = render(<FilterBox onSearch={onSearchMock}>{childComponent}</FilterBox>);

        expect(container).toHaveTextContent('My child');
    });
});
