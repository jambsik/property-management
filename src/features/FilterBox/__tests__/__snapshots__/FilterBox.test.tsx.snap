// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Box component test Should render component without errors 1`] = `
<div>
  <div
    class="MuiBox-root css-1i6msmp"
  >
    <div
      class="MuiBox-root css-iryi8b"
    >
      <div>
        My child
      </div>
      <button
        aria-label="searcheable real estate"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeMedium css-1gws2xf-MuiButtonBase-root-MuiIconButton-root"
        data-testid="filter-box-button"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
